spade-core/src/main/scala/com/salesforce/mce/spade/cli/CreateCommand.scala

```diff
@@ -2,37 +2,55 @@ package com.salesforce.mce.spade.cli
 
 import okhttp3.HttpUrl
 
-import com.salesforce.mce.spade.SpadePipeline
-import com.salesforce.mce.spade.orchard.OrchardClient
+import com.salesforce.mce.spade.SpadeWorkflowGroup
+import com.salesforce.mce.spade.orchard.{OrchardClient, OrchardClientForPipeline}
+import com.salesforce.mce.telepathy.ErrorResponse
 
-class CreateCommand(opt: CliOptions, pipeline: SpadePipeline) {
+class CreateCommand(opt: CliOptions, workflowGroup: SpadeWorkflowGroup) {
 
   def run(): Int = {
 
-    new OrchardClient(OrchardClient.Setting(HttpUrl.parse(opt.host), opt.apiKey))
-      .create(pipeline)
-      .flatMap { clientForPipeline =>
-        println(s"created workflow: ${clientForPipeline.workflowId}")
+    val (creationError, createdWorkflows) =
+      new OrchardClient(OrchardClient.Setting(HttpUrl.parse(opt.host), opt.apiKey)).create(workflowGroup)
```

Comment:
ideally we do not want the group implementation leaked to the client, client should stay simple with managing one workflow, but having an aggregator that manages all the single workflow pipeline clients.

Reply:
i think we do not have the notion of one workflow anymore. it is always workflow group. And client needs to decide how to compose the workflow groups. One workflow is a special case where group size = 1.

Reply:
got it. let's to this instead. make `.create` still return `Either`. but in this case `Right` would be `Seq` of successfully created pipelines, and `Left` would be an `ErrorResponse` plus successfully created workflows before the error. i.e. `Either[(ErrorResponse, Seq[OrchardClientForPipeline]), Seq[OrchardClientForPipeline]]`. This way the output would be much cleaner to indicate if error exist or not though the type.

Reply:
ok

Reply:
done. check again?